system_hooks/src/l1_messenger.rs

```diff
@@ -148,6 +156,9 @@ where
                     ))
                 }
             };
+            if message_offset != 32 {
```

Comment:
@StanislavBreadless in such case we are not fully compatible with solidity implementation. Do you think it's ok?